xilem_web/src/interfaces.rs

```diff
@@ -204,449 +232,2100 @@ where
     );
 }
 
-// base case for ancestor macros, do nothing, because the body is in all the child interface macros...
-#[allow(unused_macros)]
-macro_rules! for_all_element_ancestors {
-    ($($_:tt)*) => {};
+impl<State, Action, T> Element<State, Action> for T
+where
+    T: DomView<State, Action>,
+    T::Props: WithAttributes + WithClasses,
+    T::DomNode: AsRef<web_sys::Element>,
+{
 }
-#[allow(unused_imports)]
-pub(crate) use for_all_element_ancestors;
 
-macro_rules! dom_interface_macro_and_trait_definitions_impl {
-    ($interface:ident {
-        methods: $_methods_body:tt,
-        child_interfaces: {
-            $($child_interface:ident {
-                methods: $child_methods_body:tt,
-                child_interfaces: $child_interface_body: tt
-            },)*
-        }
-    }) => {
-        paste::paste! {
-            $(
-                pub trait $child_interface<T, A = ()>: $interface<T, A> $child_methods_body
-
-                /// Execute $mac which is a macro, that takes $dom_interface:ident (<optional macro parameters>) as match arm for all interfaces that
-                #[doc = concat!("`", stringify!($child_interface), "`")]
-                /// inherits from
-                macro_rules! [<for_all_ $child_interface:snake _ancestors>] {
-                    ($mac:path, $extra_params:tt) => {
-                        $mac!($interface, $extra_params);
-                        $crate::interfaces::[<for_all_ $interface:snake _ancestors>]!($mac, $extra_params);
-                    };
-                }
-                pub(crate) use [<for_all_ $child_interface:snake _ancestors>];
-            )*
-        }
-        paste::paste! {
-            /// Execute $mac which is a macro, that takes $dom_interface:ident (<optional macro parameters>) as match arm for all interfaces that inherit from
-            #[doc = concat!("`", stringify!($interface), "`")]
-            #[allow(unused_macros)]
-            macro_rules! [<for_all_ $interface:snake _descendents>] {
-                ($mac:path, $extra_params:tt) => {
-                    $(
-                        $mac!($child_interface, $extra_params);
-                        $crate::interfaces::[<for_all_ $child_interface:snake _ descendents>]!($mac, $extra_params);
-                    )*
-                };
-            }
-            #[allow(unused_imports)]
-            pub(crate) use [<for_all_ $interface:snake _descendents>];
-        }
+// #[cfg(feature = "HtmlAnchorElement")]
+pub trait HtmlAnchorElement<State, Action = ()>:
+    HtmlElement<State, Action, DomNode: AsRef<web_sys::HtmlAnchorElement>>
+{
+}
```

Comment:
This whole file is mostly generated (with a local hacky webidl based generator).
I have avoided macros because I expect this to be continuously extended overtime, and as most of us, I'm not a fan of macros, as I think DX suffers.
I've added all these commented out feature gates because I think, when some of the element props are extended (e.g. a video element has then `VideoElementProps` which composes `ElementProps` + additional stuff relevant to the video element), that inevitably the wasm binary size will grow, if we don't constrain this with features.
Because I have not yet found a way that is reasonable and robust at least to do this just with static typing, such that these additional props are optimized away when they're not needed.
I don't think it's a good idea to let all the users suffer when they don't need such features.
Currently this is not necessary because as seen below, all is basically depending on the `ElementProps` and I think the `Element` interface/trait should be included by default.